From 1b0b864a6900beeee83cadfcb596381ceb443b86 Mon Sep 17 00:00:00 2001
From: Philipp Schrader <philipp.schrader@gmail.com>
Date: Sun, 5 Mar 2023 13:51:11 -0800
Subject: [PATCH] Fix visibility of gazelle-runner

---
 def.bzl | 1 +
 1 file changed, 1 insertion(+)

diff --git a/def.bzl b/def.bzl
--- a/def.bzl
+++ b/def.bzl
@@ -225,8 +225,9 @@ def gazelle(name, testonly = False, **kwargs):
 def gazelle(name, testonly = False, **kwargs):
     runner_name, visibility = _gazelle_kwargs_prepare(name, kwargs)
 
     _gazelle_runner(
         name = runner_name,
         testonly = testonly,
+        visibility = visibility,
         **kwargs
     )
