diff --git a/BUILD.bazel b/BUILD.bazel
index e5168d0..209cc89 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -46,3 +46,37 @@ cc_library(
     strip_include_prefix = "/include",
     visibility = ["//:__subpackages__"],
 )
+
+config_setting(
+    name = "linux_arm64",
+    constraint_values = [
+        "@platforms//os:linux",
+        "@platforms//cpu:arm64",
+    ],
+)
+
+config_setting(
+    name = "linux_x86_64",
+    constraint_values = [
+        "@platforms//os:linux",
+        "@platforms//cpu:x86_64",
+    ],
+)
+
+config_setting(
+    name = "linux_arm64_msan",
+    constraint_values = [
+        "@platforms//os:linux",
+        "@platforms//cpu:arm64",
+    ],
+    values = {"features": "msan"},
+)
+
+config_setting(
+    name = "linux_x86_64_msan",
+    constraint_values = [
+        "@platforms//os:linux",
+        "@platforms//cpu:x86_64",
+    ],
+    values = {"features": "msan"},
+)
diff --git a/MODULE.bazel b/MODULE.bazel
index f8d74a9..6f93367 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -16,3 +16,6 @@ bazel_dep(name = "rules_cc", version = "0.0.10")
 bazel_dep(name = "rules_license", version = "1.0.0")
 bazel_dep(name = "rules_python", version = "0.35.0")
 bazel_dep(name = "zlib", version = "1.3")
+
+bazel_dep(name = "arm64_debian_sysroot", version = "2025.10.25")
+bazel_dep(name = "amd64_debian_sysroot", version = "2025.04.20")
diff --git a/gio/BUILD.bazel b/gio/BUILD.bazel
index ac78a44..7779117 100644
--- a/gio/BUILD.bazel
+++ b/gio/BUILD.bazel
@@ -625,7 +625,7 @@ REPO_ROOT = package_relative_label(":BUILD.bazel").workspace_root
 INCLUDE_PREFIX = REPO_ROOT if REPO_ROOT else "."
 
 cc_library(
-    name = "gio",
+    name = "gio_src",
     srcs = GIO_BASE_SOURCES + select({
         "@platforms//os:linux": ["gcontenttype-fdo.c"],
         "@platforms//os:macos": [],
@@ -711,3 +711,15 @@ cc_library(
         "//gobject",
     ],
 )
+
+alias(
+    name = "gio",
+    actual = select({
+        "//:linux_arm64_msan": ":gio_src",
+        "//:linux_x86_64_msan": ":gio_src",
+        "//:linux_arm64": "@arm64_debian_sysroot//:gio-2.0",
+        "//:linux_x86_64": "@amd64_debian_sysroot//:gio-2.0",
+        "//conditions:default": ":gio_src",
+    }),
+    visibility = ["//visibility:public"],
+)
diff --git a/glib/BUILD.bazel b/glib/BUILD.bazel
index b5f457d..863d0d9 100644
--- a/glib/BUILD.bazel
+++ b/glib/BUILD.bazel
@@ -257,7 +257,7 @@ REPO_ROOT = package_relative_label(":BUILD.bazel").workspace_root
 INCLUDE_PREFIX = REPO_ROOT if REPO_ROOT else "."
 
 cc_library(
-    name = "glib",
+    name = "glib_src",
     srcs = GLIB_SRCS + select({
         "@platforms//os:windows": GLIB_WIN32_SRCS,
         "//conditions:default": GLIB_POSIX_SRCS,
@@ -315,3 +315,15 @@ cc_library(
     ],
     visibility = ["//visibility:public"],
 )
+
+alias(
+    name = "glib",
+    actual = select({
+        "//:linux_arm64_msan": ":glib_src",
+        "//:linux_x86_64_msan": ":glib_src",
+        "//:linux_arm64": "@arm64_debian_sysroot//:glib-2.0",
+        "//:linux_x86_64": "@amd64_debian_sysroot//:glib-2.0",
+        "//conditions:default": ":glib_src",
+    }),
+    visibility = ["//visibility:public"],
+)
diff --git a/gobject/BUILD.bazel b/gobject/BUILD.bazel
index eb2cda2..47b678d 100644
--- a/gobject/BUILD.bazel
+++ b/gobject/BUILD.bazel
@@ -127,7 +127,7 @@ REPO_ROOT = package_relative_label(":BUILD.bazel").workspace_root
 INCLUDE_PREFIX = REPO_ROOT if REPO_ROOT else "."
 
 cc_library(
-    name = "gobject",
+    name = "gobject_src",
     srcs = GOBJECT_SRCS + PUBLIC_HDRS + [
         "gatomicarray.h",
         "glib-enumtypes.c",
@@ -156,3 +156,19 @@ cc_library(
+    target_compatible_with = select({
+        "@platforms//cpu:armv7": ["@platforms//:incompatible"],
+        "//conditions:default": [],
+    }),
     visibility = ["//visibility:public"],
     deps = ["//glib"],
 )
+
+alias(
+    name = "gobject",
+    actual = select({
+        "//:linux_arm64_msan": ":gobject_src",
+        "//:linux_x86_64_msan": ":gobject_src",
+        "//:linux_arm64": "@arm64_debian_sysroot//:gobject-2.0",
+        "//:linux_x86_64": "@amd64_debian_sysroot//:gobject-2.0",
+        "//conditions:default": ":gobject_src",
+    }),
+    visibility = ["//visibility:public"],
+)
