diff --git a/BUILD.bazel b/BUILD.bazel
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -47,3 +47,37 @@
     strip_include_prefix = "/include",
     visibility = ["//:__subpackages__"],
 )
+
+config_setting(
+    name = "linux_arm64",
+    constraint_values = [
+        "@platforms//os:linux",
+        "@platforms//cpu:arm64",
+    ],
+)
+
+config_setting(
+    name = "linux_x86_64",
+    constraint_values = [
+        "@platforms//os:linux",
+        "@platforms//cpu:x86_64",
+    ],
+)
+
+config_setting(
+    name = "linux_arm64_msan",
+    constraint_values = [
+        "@platforms//os:linux",
+        "@platforms//cpu:arm64",
+    ],
+    values = {"features": "msan"},
+)
+
+config_setting(
+    name = "linux_x86_64_msan",
+    constraint_values = [
+        "@platforms//os:linux",
+        "@platforms//cpu:x86_64",
+    ],
+    values = {"features": "msan"},
+)
diff --git a/MODULE.bazel b/MODULE.bazel
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -17,3 +17,6 @@
 bazel_dep(name = "rules_license", version = "1.0.0")
 bazel_dep(name = "rules_python", version = "0.35.0")
 bazel_dep(name = "zlib", version = "1.3")
+
+bazel_dep(name = "arm64_debian_sysroot", version = "2025.10.25")
+bazel_dep(name = "amd64_debian_sysroot", version = "2025.04.20")
diff --git a/gio/BUILD.bazel b/gio/BUILD.bazel
--- a/gio/BUILD.bazel
+++ b/gio/BUILD.bazel
@@ -750,7 +750,7 @@
 )
 
 cc_library(
-    name = "gio",
+    name = "gio_src",
     srcs = GIO_BASE_SOURCES + select({
         "@platforms//os:linux": ["gcontenttype-fdo.c"],
         "@platforms//os:macos": [],
@@ -913,3 +913,15 @@
         "//conditions:default": ["//glib:stub_intl"],
     }),
 )
+
+alias(
+    name = "gio",
+    actual = select({
+        "//:linux_arm64_msan": ":gio_src",
+        "//:linux_x86_64_msan": ":gio_src",
+        "//:linux_arm64": "@arm64_debian_sysroot//:gio-2.0",
+        "//:linux_x86_64": "@amd64_debian_sysroot//:gio-2.0",
+        "//conditions:default": ":gio_src",
+    }),
+    visibility = ["//visibility:public"],
+)
diff --git a/glib/BUILD.bazel b/glib/BUILD.bazel
--- a/glib/BUILD.bazel
+++ b/glib/BUILD.bazel
@@ -322,7 +322,7 @@
 INCLUDE_PREFIX = REPO_ROOT if REPO_ROOT else "."
 
 cc_library(
-    name = "glib",
+    name = "glib_src",
     srcs = GLIB_SRCS + select({
         "@platforms//os:windows": GLIB_WIN32_SRCS,
         "//conditions:default": GLIB_POSIX_SRCS,
@@ -450,3 +450,15 @@
     "",
     "-console",
 ]]
+
+alias(
+    name = "glib",
+    actual = select({
+        "//:linux_arm64_msan": ":glib_src",
+        "//:linux_x86_64_msan": ":glib_src",
+        "//:linux_arm64": "@arm64_debian_sysroot//:glib-2.0",
+        "//:linux_x86_64": "@amd64_debian_sysroot//:glib-2.0",
+        "//conditions:default": ":glib_src",
+    }),
+    visibility = ["//visibility:public"],
+)
diff --git a/gobject/BUILD.bazel b/gobject/BUILD.bazel
--- a/gobject/BUILD.bazel
+++ b/gobject/BUILD.bazel
@@ -138,7 +138,7 @@
 INCLUDE_PREFIX = REPO_ROOT if REPO_ROOT else "."
 
 cc_library(
-    name = "gobject",
+    name = "gobject_src",
     srcs = GOBJECT_SRCS + PUBLIC_HDRS + [
         "gatomicarray.h",
         "glib-enumtypes.c",
@@ -164,6 +164,22 @@
         "G_LOG_DOMAIN='\"GLib-GObject\"'",
         "GOBJECT_COMPILATION",
     ],
+    target_compatible_with = select({
+        "@platforms//cpu:armv7": ["@platforms//:incompatible"],
+        "//conditions:default": [],
+    }),
     visibility = ["//visibility:public"],
     deps = ["//glib"],
 )
+
+alias(
+    name = "gobject",
+    actual = select({
+        "//:linux_arm64_msan": ":gobject_src",
+        "//:linux_x86_64_msan": ":gobject_src",
+        "//:linux_arm64": "@arm64_debian_sysroot//:gobject-2.0",
+        "//:linux_x86_64": "@amd64_debian_sysroot//:gobject-2.0",
+        "//conditions:default": ":gobject_src",
+    }),
+    visibility = ["//visibility:public"],
+)
